modules/selection/main/index.coffee

```diff
@@ -16,9 +16,41 @@ getMethod = (node, methodName) ->
   else
     Element.prototype[methodName]
 
+getMatches = (node) ->
+  # XXX: use hx.vendor
+  matchesMethod =
+    if node.matches then 'matches'
+    else if node.matchesSelector then 'matchesSelector'
+    else if node.mozMatchesSelector then 'mozMatchesSelector'
+    else if node.webkitMatchesSelector then 'webkitMatchesSelector'
+    else if node.oMatchesSelector then 'oMatchesSelector'
+    else if node.msMatchesSelector then 'msMatchesSelector'
+
+  if matchesMethod then getMethod(node, matchesMethod)
+  else (selector) ->
+    matchingNodes = getMethod(this, 'querySelector').call(this, selector)
+    i = matchingNodes.length
+    i while --i >= 0 and matchingNodes.item(i) isnt this
```

Comment:
The 'polyfill' was taken from the MDN which is why it's a bit odd:
```js
​function matches(elm, selector) {
    var matches = (elm.document || elm.ownerDocument).querySelectorAll(selector),
        i = matches.length;
    while (--i >= 0 && matches.item(i) !== elm) {}
    return i > -1;
}
```

Looking at it, the implementation is doing something different from the polyfill anyway (which is wrong and won't work)